src/java/org/apache/cassandra/utils/RMICloseableSocketFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils;
+
+import java.io.IOException;
+
+/**
+ * This represents closeable RMI Socket factory. It could be implemented for client and/or server socket factories.
+ */
+public interface RMICloseableSocketFactory
```

Comment:
@maulin-vasavada you can rework this to abstract class extending `RMISocketFactory` and then all current implementations extending `RMISocketFactory` would extend `RMICloseableSocketFactory` instead. That way you can set clientFactory below to be RMICloseableSocketFactory and you do not need to cast anything.

Reply:
Will look into it.

Reply:
There is no common ground between `RMIClientSocketFactory` and `RMIServerSocketFactory` from `java.rmi.server`. So to follow your suggestion I'd have to create `RMICloseableClientSocketFactory` and `RMICloesableServerSocketFactory` interfaces and then ultimately avoid the casting you are pointing to.